src/main/java/org/apache/sling/installer/provider/file/impl/FileInstaller.java

```diff
@@ -84,6 +88,13 @@ public void start(final OsgiInstaller installer, final SlingSettingsService sett
                 key = "${sling.home}" + key.substring(settings.getSlingHomePath().length());
             }
             logger.debug("Starting monitor for {}", config.directory);
+            if (autoCreateDirectories) {
+                try {
+                    Files.createDirectory(Paths.get(config.directory));
```

Comment:
shouldn't we use here createDirectories instead of createDirectory? not tested it, but just looking at the [documentation](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createDirectories-java.nio.file.Path-java.nio.file.attribute.FileAttribute...-) it seems that the latter may create an exception if the directory already exists, whereas the former should not. furthermore it might be helpful to also create missing parent directories.

Reply:
Yes indeed, logs 
```
22.11.2021 16:20:53.877 *WARN* [FelixStartLevel] org.apache.sling.installer.provider.file.impl.FileInstaller Could not create directory to monitor at /Users/konradwindszus/git/sling/sling-org-apache-sling-starter/target/launchers/sling-12-oak-tar/install
java.nio.file.FileAlreadyExistsException: /Users/konradwindszus/git/sling/sling-org-apache-sling-starter/target/launchers/sling-12-oak-tar/install
...
```

in case the directory does already exist

Reply:
Fixed in https://github.com/apache/sling-org-apache-sling-installer-provider-file/commit/6bbc945bb543de07245b78c983214148ef43c212.